### ISSUE TITLE
float-compare: false negative when comparing to a non-float

### ISSUE THREAD

**Issue by scop** [Antonboom/testifylint#24]
```go
	var foo int // or `any`, or seemingly any non-float type 
	foo = 3
	assert.Equal(t, float64(3), foo)
```
I think this should be flagged by `float-compare`, but is not at the moment.

Essentially, if `expected` is a float of some kind, I think it should be flagged by `float-compare`, no matter what `actual` is.

**Comment by Antonboom** [comment]
@scop, hi!

Offtop for this issue, but what do you think, should `float-compare` ignores comparing with `0`?

```go
assert.Equal(t, float64(0), actual)
```

🤔

**Comment by scop** [comment]
> Offtop for this issue, but what do you think, should `float-compare` ignores comparing with `0`?

If comparing against `float64(0)` has the same problems as comparing against other float values, then I don't think it should be ignored.

**Comment by Antonboom** [comment]
@scop yep, I just do not remember code like this

```go
assert.InEpsilon(t, 0., fl, 0.)	
```

instead of 

```go
assert.Equal(t, 0., fl)
```
